### ISSUE TITLE
Support id as parameter in get_animals()

### ISSUE THREAD

**Issue by peterdesmet** [inbo/etn#74]
I would like `get_animals()` to have an additional parameter `id`, to get metadata on specific animals I look for.

**Comment by peterdesmet** [comment]
@jreubens @IPauwels @PieterjanVerhelst, the following etn functions allow to search for a single item by ID:

- `get_receivers()`: possible, with parameter `receiver_id`
- `get_tags()`: possible, with parameter `tag_id`
- `get_projects()`: not possible, not necessary
- `get_deployments()`: not possible, probably not necessary
- `get_detections()`: not possible, not necessary
- `get_animals()`: **not possible, would be nice to have**

For consistency, `get_animals()` would have the parameter `animal_id`, but that field is currently only used for 233 of the 1,956 animals (12%) and not all are unique.

Should we aim to promote the use the researcher assigned `animal_id` as unique animal identifiers (cf receiver and tag) or do you think we should rely on the database-assigned `animal_fk` instead?

In the later case, I would be tempted to rename/duplicate `pk` to `animal_id` and move the content from the researcher-assigned one to the currently not used `animal_nickname`.

Feedback welcome!

**Comment by peterdesmet** [comment]
## Current situation

pk | animal_id | animal_project_code | tag_id | tag_fk | animal_nickname
--- | --- | --- | --- | --- | ---
234 | | 2011_rivierprik | A69-1303-3556 | 213 | 
235 | A1 | 2015_phd_verhelst | A69-1601-13631 | 405 | 

Disadvantage: `animal_id` cannot be used as a parameter, as it is not unique or always populated. `pk` is the unique identifier, but that name is not consistent with `tag_id` or `receiver_id` and when publishing it to the outside world, it will be promoted as an `animal_id`, which will be confusing with the current field `animal_id`

## Suggestion

pk | animal_id | animal_project_code | tag_id | tag_fk | animal_nickname
--- | --- | --- | --- | --- | ---
234 | 234 | 2011_rivierprik | A69-1303-3556 | 213 | 
235 | 235 | 2015_phd_verhelst | A69-1601-13631 | 405 | A1

`pk` = `animal_id`: will always be unique and always be populated. It can be promoted as `animal_id` to the outside world and as a function parameter. Researcher can still assign their own name/code in in `animal_nickname`, but are not required to.

**Comment by PieterjanVerhelst** [comment]
Indeed, I would suggest as well to use the database unique key as the animal_id :thumbsup:

**Comment by IPauwels** [comment]
@peterdesmet I support your suggestion
@PieterjanVerhelst yours as well to enable the indication of an additional lable ID. Perhaps the latter can be named label_id to be in line with "tag_id"? Or do we have the possibility (and is it a good idea or not?) to view an additional plastic label as a tag? 

Then a question regarding this situation: if you tag an animal with two tags (not plastic ones, but for instance an acoustic tag + a PIT tag), do you include the animal twice in the database and each time with the other tag_id?

**Comment by peterdesmet** [comment]
Ok, so I see consensus for using the database pk as the identifier for the animal, and naming it “animal_id”

But not for moving A1 etc to animal_nickname. I like animal_label better than label_id (which seems to imply another label table?).

Do you think animal_nickname will be used at all?

And animals with two tags (consecutive in time, or at the same time) appear twice in the table, but we update the pkg code, so that info is collapsed and only one animal per row is returned: #106

**Comment by PieterjanVerhelst** [comment]
I think animal_nickname will be used; for twaite shad we did this, but I don't think it was fed into the database.
@IPauwels, correct remark about double tagging. I prefer various tag information is given in the same animal/line. Would it be an option to make a distinction by different tag IDs, by including the telemetry type in the header's name:
- acoustic_tag_id
- pit_tag_id
- radio_tag_id
- dst_tag_id
- psat_tag_id
These are the most common applied telemetry techniques.

Another note: on the ETN database you can switch between different telemetry techniques on the top left of the website; now you have the choice between acoustic and c-pods. Hence, the acoustic_tag_id field should be the default and only mandatory field.
@jreubens , what is your opinion on this?

**Comment by peterdesmet** [comment]
@PieterjanVerhelst multiple tags associated with one animal seem to currently be of the same type: 
fk | animal_id | animal_project_code | tag_id | tag_fk | scientific_name
-- | -- | -- | -- | -- | --
2834 | NA | 2015_phd_verhelst | A69-9006-917,A69-9006-918 | 1807,1808 | Gadus morhua

Where is this other tag (different kind) information stored?

**Comment by PieterjanVerhelst** [comment]
@peterdesmet not really. Those two codes are both acoustic codes. Specifically, acoustic transmitters with sensors have multiple codes. For instance, a transmitter with a temperature and depth sensor has two IDs: one for the temperature information and one for the pressure. Yet, it is the same device/transmitter. When uploading an animal in the ETN database, the `tag_id` field uses a drop downs field to select the respective transmitter linked to the fish. Hence, in this `tag_id` field, it is not possible to add different transmitters of a different type, since you can only select one. If this explanation is not clear, feel free to call me.

**Comment by peterdesmet** [comment]
Ok, so `A69-9006-917`, `A69-9006-918` are two different sensors in the same acoustic device/transmitter. 👍 

Do you have examples currently in the database of non-acoustic tags? Such pit tags or radio tags? And where are those tag_ids stored?

**Comment by PieterjanVerhelst** [comment]
Indeed, that is correct.
I myself don't at the moment. However, I have eels in the database with both an acoustic and pop-off tag, but only the acoustic tag is indicated because there is no field yet for the pop-off tag ID.  @IPauwels did you store PIT IDs in the database? I would assume this is added to the `comments` field of the `animal` information

**Comment by damianooldoni** [comment]
Based on @PieterjanVerhelst's comment, I think `animal_nickname` is not ideal as it is not what people imagine while thinking to nicknames. `animal_label` sounds me much better.

**Comment by peterdesmet** [comment]
Ok, since the discussion is diverging here, I'll split this in two issues

- Including other tag ids is now a new issue: #108
- Changing the view to use animal id is now a new issue: #109
- This issue reserved for adding the `animal_id` parameter in the `get_animals()` function

**Comment by peterdesmet** [comment]
Issue blocked by https://github.com/inbo/etn/issues/109

**Comment by peterdesmet** [comment]
Fixed in #116
